tests/gateway/providers/test_bedrock.py

```diff
@@ -0,0 +1,480 @@
+        )
+        provider.get_bedrock_client()
+
+        if "aws_region" in aws_config:
+            _assert_any_call_at_least(mock_session, region_name=aws_config["aws_region"])
+
+        if "aws_role_arn" in aws_config:
+            _assert_any_call_at_least(mock_client, service_name="sts")
+            _assert_any_call_at_least(mock_assume_role, RoleArn=aws_config["aws_role_arn"])
+            _assert_any_call_at_least(mock_client, service_name="bedrock")
+
+        elif {"aws_secret_access_key", "aws_access_key_id"} <= set(aws_config):
+            _assert_any_call_at_least(mock_client, service_name="bedrock")
+            _assert_any_call_at_least(
+                mock_client,
+                **{
+                    k: v
+                    for k, v in aws_config.items()
+                    if k in {"aws_secret_access_key", "aws_access_key_id"}
+                },
+            )
+
+
+@pytest.mark.asyncio
+@pytest.mark.parametrize("aws_config", [c[0] for c in bedrock_aws_configs])
+@pytest.mark.parametrize(
+    "provider,config,payload,response,expected",
+    [
+        pytest.param(
+            fix["provider"],
+            fix["config"],
+            fix["request"],
+            fix["response"],
+            fix["expected"],
+            marks=[]
+            if fix["provider"] is not AWSBedrockModelProvider.COHERE
+            else pytest.mark.skip("Cohere isn't availabe on AWS Bedrock yet"),
+        )
+        for fix in bedrock_model_provider_fixtures
+    ],
+)
+async def test_bedrock_request_response(provider, config, payload, response, expected, aws_config):
+    with mock.patch(
+        "mlflow.gateway.providers.bedrock.AWSBedrockProvider._make_request", return_value=response
+    ) as mock_post:
+        if not expected:
+            pytest.skip("no expected value")
+        expected["metadata"]["model"] = config["model"]["name"]
+        provider = AWSBedrockProvider(
+            RouteConfig(**_merge_model_and_aws_config(config, aws_config))
+        )
+        response = await provider.completions(completions.RequestPayload(**payload))
+        assert jsonable_encoder(response) == expected
+        mock_post.assert_called_once()
+
+
+@pytest.mark.skipif(
+    "MLFLOW_AI_GATEWAY_LIVE_TEST_BEDROCK" not in os.environ, reason="don't run live tests"
+)
+@pytest.mark.asyncio
+@pytest.mark.parametrize("aws_config", bedrock_live_aws_configs)
+@pytest.mark.parametrize(
+    "provider,config,payload,expected",
+    [
+        pytest.param(
+            fix["provider"],
+            fix["config"],
+            fix["request"],
+            fix["expected"],
+            marks=[]
+            if fix["provider"] is not AWSBedrockModelProvider.COHERE
+            else pytest.mark.skip("Cohere isn't availabe on AWS Bedrock yet"),
+        )
+        for fix in bedrock_model_provider_fixtures
+    ],
+)
+async def test_live_call_to_aws(provider, config, payload, expected, aws_config):
```

Comment:
This test should be removed to `test_integration.py` and the outside call to AWS should be mocked. We're not trying to test that AWS can return the results from the language model services, but rather that the client and fluent APIs are capable of communicating up to the point of calling and receiving a response from a provider (in this case, AWS). 
Please have a look at the server constructor fixtures referenced within `test_integration` and see how we have mocked the external calls to other providers for guidance.